Library/src/Services/WebService.cpp

```diff
@@ -104,9 +105,16 @@ void ResultBase::SetResult(csp::services::EResultCode ResCode, uint16_t HttpResC
 	HttpResponseCode = HttpResCode;
 }
 
-int ResultBase::ParseErrorCode(const csp::common::String& Value)
+EResultBaseFailureReason ResultBase::ParseErrorCode(const csp::common::String& Value)
 {
-	return (int) EResultBaseFailureReason::Unknown;
+	if (XErrorCodeToFailureReason.HasKey(Value))
+	{
+		return XErrorCodeToFailureReason[Value];
+	}
+	else
+	{
+		return EResultBaseFailureReason::Unknown;
```

Comment:
Worth including a log message here to help us catch error codes that we don't yet track in CSP?

Reply:
Saw the new error message. Super useful! Thanks :)